package.json

```diff
@@ -2,15 +2,28 @@
   "name": "minio",
   "version": "7.0.33",
   "description": "S3 Compatible Cloud Storage client",
-  "main": "./dist/main/minio.js",
+  "main": "dist/main/minio.cjs",
+  "module": "dist/main/minio.mjs",
+  "types": "dist/main/minio.d.ts",
+  "browser": "dist/main/minio.iife.js",
+  "jsdelivr": "dist/main/minio.iife.js",
+  "unpkg": "dist/main/minio.iife.js",
+  "exports": {
+    ".": {
+      "import": "./dist/main/minio.mjs",
+      "require": "./dist/main/minio.cjs"
+    },
+    "./package.json": "./package.json",
+    "./*": "./*"
+  },
   "scripts": {
-    "compile": "gulp compile",
-    "test": "gulp test",
-    "lint": "gulp lint",
-    "lint-fix": "gulp lint --fix",
-    "prepublishOnly": "gulp test",
-    "functional": "gulp functionalTest",
-    "browserify": "gulp browserify",
+    "compile": "rollup -c",
+    "test": "mocha test/**/*.js",
+    "lint": "eslint --ignore-path .gitignore .",
+    "lint-fix": "eslint --ignore-path .gitignore --fix .",
+    "prepublishOnly": "npm run test",
+    "functional": "mocha test/functional/*.js ",
+    "browserify": "npm run compile",
```

Comment:
Removing gulp, and replacing it with [rollup](https://rollupjs.org/) just for the build step. For everything else, we can directly run their CLI commands (mocha, eslint).

Rollup itself isn't a shiny bundler. A shiny bundler would be [esbuild](https://github.com/evanw/esbuild), [bun](https://github.com/oven-sh/bun) or [turbo](https://github.com/vercel/turbo).